Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM gradle:8.2.1-jdk17 as cache
+RUN mkdir -p /home/gradle/cache_home
+ENV GRADLE_USER_HOME /home/gradle/cache_home
+COPY build.gradle.kts /home/gradle/java-code/
+WORKDIR /home/gradle/java-code
+RUN gradle clean build -i --stacktrace
+
+FROM gradle:8.2.1-jdk17 as builder
+COPY --from=cache /home/gradle/cache_home /home/gradle/.gradle
+COPY . /usr/src/java-code/
+WORKDIR /usr/src/java-code
+RUN gradle bootJar -i --stacktrace
+
+FROM openjdk:17-slim
+EXPOSE 8080
+WORKDIR /usr/src/java-app
+COPY --from=builder /usr/src/java-code/build/libs/*.jar ./app.jar
+ENTRYPOINT ["java", "-jar", "app.jar"]
```

Comment:
how much time does it takes to build the image?

Reply:
Initial pure build : docker-compose down && docker system prune -a -f && time docker-compose build
Result : docker-compose build  0.98s user 1.42s system 3% cpu 1:19.41 total
docker-compose up --build --force-recreate
Result : [+] Building 16.1s (18/18) FINISHED 
docker-compose up --build   
Result : [+] Building 0.8s (18/18) FINISHED

Reply:
@ajax-gnatiuk-s

Reply:
It's too long. If it ok for you then leave it as is.

A few seconds is acceptable time for me about. But it's up to you. If you like wasting time on waiting while it builds, it's your choice ;)

In general I like your experiments, but let's follow mentor's advices. It's really nice that you know how to cook multistage builds but building the app inside the docker doesn't make any sense.

Reply:
Nevertheless, I approve this PR